presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergSplitManager.java

```diff
@@ -0,0 +1,249 @@
+        // There should be 2 splits after file scanning with the filter
+        validateSplitsPlannedForSql(splitManager, transactionManager, filterPushdown,
+                "select * from " + tableName + " where a = 'var1' and r.c = 1008",
+                2);
+
+        assertQuerySucceeds("DROP TABLE " + tableName);
+    }
+
+    private void testGetSplitsByNonIdentityPartitionColumns(String tableName, boolean filterPushdown)
+    {
+        // Take transformer 'year()' for example
+        assertUpdate("CREATE TABLE " + tableName + " (a date, b integer, r row(c int, d varchar)) WITH(partitioning = ARRAY['year(a)'])");
+
+        // Firstly, we build an Iceberg table which has three files:
+        //      file1: partition column a's value is year(1984) with range ('1984-01-08', '1984-07-08'), and column b's value between (1, 3)
+        //      file2: partition column a's value is year(2001) with range ('2001-01-01', '2001-11-01'), and column b's value between (8, 10)
+        //      flie3: partition column a's value is year(1984) with range ('1984-06-08', '1984-12-08'), and column b's value between (2, 9)
+        assertUpdate("INSERT INTO " + tableName + " VALUES (date '1984-01-08', 1, (1001, 't1')), (date '1984-07-08', 3, (1003, 't3'))", 2);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (date '2001-01-01', 8, (1008, 't8')), (date '2001-11-01', 10, (1010, 't10'))", 2);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (date '1984-06-08', 2, (1002, 't2')), (date '1984-12-08', 9, (1009, 't9'))", 2);
+
+        TransactionManager transactionManager = getQueryRunner().getTransactionManager();
+        SplitManager splitManager = getQueryRunner().getSplitManager();
+
+        // Get a plan for query on <tableName> with filter `a = date '1984-03-01'`
+        // There should leave only 1 split after file scanning with the filter
+        validateSplitsPlannedForSql(splitManager, transactionManager, filterPushdown,
+                "select * from " + tableName + " where a = date '1984-03-01'",
+                1);
+
+        // Get a plan for query on <tableName> with filter `a = date '1984-06-09'`
+        // There should be 2 splits after file scanning with the filter
+        validateSplitsPlannedForSql(splitManager, transactionManager, filterPushdown,
+                "select * from " + tableName + " where a = date '1984-06-09'",
+                2);
+
+        // Get a plan for query on <tableName> with filter `a = date '1984-06-09' and b = 8`
+        // There should leave only 1 split after file scanning with the filter
+        validateSplitsPlannedForSql(splitManager, transactionManager, filterPushdown,
+                "select * from " + tableName + " where a = date '1984-06-09' and b = 8",
+                1);
+
+        assertQuerySucceeds("DROP TABLE " + tableName);
+    }
+
+    private Session withPushdownFilter(boolean pushdown)
```

Comment:
nit: I think we can name this as sessionWithFilterPushdown?

Reply:
Ok, done!